experiments/yarn/src/main/scala/io/gearpump/experiments/yarn/client/Client.scala

```diff
@@ -17,323 +17,171 @@
-- [ ] Configuration file needs to specify minimum master ram, vmcore requirements
-- [ ] Configuration file needs to specify minimum worker ram, vmcore requirements
-- [ ] Configuration file should specify where in HDFS to place jars for appmaster and workers
-- [ ] Client needs to use YARN cluster API to find best nodes to run Master(s)
-- [ ] Client needs to use YARN cluster API to find best nodes to run Workers
- */
+  private val LOG: Logger = LogUtil.getLogger(getClass)
 
-trait ClientAPI {
-  def getConfiguration: AppConfig
-  def getCommand: String
-  def getYarnConf: YarnConfiguration
-  def getAppEnv: Map[String, String]
-  def getAMCapability: Resource
-  def waitForTerminalState(appId: ApplicationId): ApplicationReport
-  def start(): Boolean
-  def submit(): Try[ApplicationId]
-}
+  private val queue = akka.getString(APPMASTER_QUEUE)
+  private val memory = akka.getString(APPMASTER_MEMORY).toInt
+  private val vcore = akka.getString(APPMASTER_VCORES).toInt
 
-class Client(configuration: AppConfig, yarnConf: YarnConfiguration, yarnClient: YarnClient,
-             containerLaunchContext: (String) => ContainerLaunchContext, fileSystem: FileSystem) extends ClientAPI {
-  import Client._
-  import yarn.Constants._
+  private val version = Util.version
 
-  private val LOG: Logger = LogUtil.getLogger(getClass)
-  def getConfiguration = configuration
-  def getYarnConf = yarnConf
-  def getFs = FileSystem.get(getYarnConf)
-  private def getEnv = getConfiguration.getEnv _
-
-  private val version = configuration.getEnv("version")
-  private val confOnYarn = getEnv(HDFS_ROOT) + "/conf/"
-  private val restURI = s"api/$REST_VERSION"
-
-
-  private[client] def getMemory(envVar: String): Int = {
-    try {
-      getEnv(envVar).trim.toInt
-    } catch {
-      case throwable: Throwable =>
-        MEMORY_DEFAULT
-    }
-  }
+  def submit(): ApplicationReport = {
+    LOG.info("Starting AM")
 
-  def getCommand = {
-    val exe = getEnv(YARNAPPMASTER_COMMAND)
-    val classPath = Array(
-      s"pack/$version/conf",
-      s"pack/$version/dashboard",
-      s"pack/$version/lib/*",
-      s"pack/$version/lib/daemon/*",
-      s"pack/$version/lib/services/*",
-      s"pack/$version/lib/yarn/*",
-      "yarnConf"
-    )
-    val mainClass = getEnv(YARNAPPMASTER_MAIN)
-    val logdir = ApplicationConstants.LOG_DIR_EXPANSION_VAR
-    val command = s"$exe  -cp ${classPath.mkString(File.pathSeparator)}${File.pathSeparator}" +
-      "$CLASSPATH" +
-      s" -D${Constants.GEARPUMP_HOME}=${Environment.LOCAL_DIRS.$$()}/${Environment.CONTAINER_ID.$$()}/pack" +
-      s" $mainClass" +
-      s" -version $version" +
-      " 1>" + logdir +"/" + ApplicationConstants.STDOUT +
-      " 2>" + logdir +"/" + ApplicationConstants.STDERR
-
-    LOG.info(s"command=$command")
-    command
-  }
+    // first step, check the version, to make sure local version matches remote version
+    val rootEntry = rootEntryPath(zip = gearPackagePath)
```

Comment:
This used to work with package files of *.tar.gz. We should be able to handle either *.zip or *.tar.gz.